build.sh

```diff
@@ -587,6 +591,63 @@ run_rust_tests() {
   fi
 }
 
+#-----------------------------------------------------------------------------
+# Function: run_rust_valgrind_tests
+# Run Rust Valgrind tests (to detect memory leaks)
+#-----------------------------------------------------------------------------
+run_rust_valgrind_tests() {
+  if [[ "$RUN_RUST_VALGRIND" != "1" ]]; then
+    return 0
+  fi
+
+  echo "Running Rust tests..."
+
+  # Set Rust test environment
+  RUST_DIR="$ROOT/src/redisearch_rs"
+
+  # Set up RUSTFLAGS for warnings
+  if [[ "$RUST_DENY_WARNS" == "1" ]]; then
+    export RUSTFLAGS="${RUSTFLAGS:+${RUSTFLAGS} }-D warnings"
+  fi
+
+  # Pin a specific working version of nightly to prevent breaking the CI because
+  # regressions in a nightly build.
+  # Make sure to synchronize updates across all modules: Redis and RedisJSON.
+  NIGHTLY_VERSION="nightly-2025-07-30"
```

Comment:
This should not be duplicated. The whole point of this var was to have the nightly version defined in a single place.